From 09b86cc0b08412b5239657aad2cc0780f1839b12 Mon Sep 17 00:00:00 2001
From: Yinbo Zhu <yinbo.zhu@nxp.com>
Date: Wed, 26 Sep 2018 10:42:17 +0800
Subject: [PATCH 606/767] Revert "usb: dwc3: Implement workaround for Erratum
 A009116"

This reverts commit 5431996fdbcc5f6fe69de087cc6d1d0459ca71dc.
[Xulin: Original patch taken from NXP LSDK-18.12.]
Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/usb/dwc3/core.c | 22 ----------------------
 drivers/usb/dwc3/core.h |  7 -------
 2 files changed, 29 deletions(-)

diff --git a/drivers/usb/dwc3/core.c b/drivers/usb/dwc3/core.c
index c9c6a537445f..a0c606e07d3d 100644
--- a/drivers/usb/dwc3/core.c
+++ b/drivers/usb/dwc3/core.c
@@ -74,11 +74,6 @@ static int dwc3_get_dr_mode(struct dwc3 *dwc)
 		mode = USB_DR_MODE_HOST;
 		break;
 	default:
-	       /* Adjust Frame Length */
-		if (dwc->configure_gfladj)
-		dwc3_writel(dwc->regs, DWC3_GFLADJ, GFLADJ_30MHZ_REG_SEL |
-				GFLADJ_30MHZ(GFLADJ_30MHZ_DEFAULT));
-
 		if (IS_ENABLED(CONFIG_USB_DWC3_HOST))
 			mode = USB_DR_MODE_HOST;
 		else if (IS_ENABLED(CONFIG_USB_DWC3_GADGET))
@@ -1201,12 +1196,6 @@ static void dwc3_get_properties(struct dwc3 *dwc)
 	dwc->dma_coherent = device_property_read_bool(dev,
 				"dma-coherent");
 
-	dwc->needs_fifo_resize = of_property_read_bool(node, "tx-fifo-resize");
-
-	dwc->configure_gfladj =
-			of_property_read_bool(node, "configure-gfladj");
-	dwc->dr_mode = of_usb_get_dr_mode(node);
-	
 	device_property_read_u8(dev, "snps,rx-thr-num-pkt-prd",
 				&rx_thr_num_pkt_prd);
 	device_property_read_u8(dev, "snps,rx-max-burst-prd",
@@ -1350,7 +1339,6 @@ static int dwc3_probe(struct platform_device *pdev)
 
 	void __iomem		*regs;
 
-	struct device_node      *node = dev->of_node;
 	dwc = devm_kzalloc(dev, sizeof(*dwc), GFP_KERNEL);
 	if (!dwc)
 		return -ENOMEM;
@@ -1374,11 +1362,6 @@ static int dwc3_probe(struct platform_device *pdev)
 	dwc->xhci_resources[0].flags = res->flags;
 	dwc->xhci_resources[0].name = res->name;
 
-	if (node) {
-		dwc->configure_gfladj =
-			of_property_read_bool(node, "configure-gfladj");
-	}
-
 	/*
 	 * Request memory region but exclude xHCI regs,
 	 * since it will be requested by the xhci-plat driver.
@@ -1438,11 +1421,6 @@ static int dwc3_probe(struct platform_device *pdev)
 	if (ret < 0)
 		goto err1;
 
-	/* Adjust Frame Length */
-	if (dwc->configure_gfladj)
-	dwc3_writel(dwc->regs, DWC3_GFLADJ, GFLADJ_30MHZ_REG_SEL |
-		    GFLADJ_30MHZ(GFLADJ_30MHZ_DEFAULT));
-
 	pm_runtime_forbid(dev);
 
 	ret = dwc3_alloc_event_buffers(dwc, DWC3_EVENT_BUFFERS_SIZE);
diff --git a/drivers/usb/dwc3/core.h b/drivers/usb/dwc3/core.h
index 8e1c64ac6db9..ac9538641c27 100644
--- a/drivers/usb/dwc3/core.h
+++ b/drivers/usb/dwc3/core.h
@@ -360,11 +360,6 @@
 #define DWC3_GHWPARAMS6_SRPSUPPORT		BIT(10)
 #define DWC3_GHWPARAMS6_EN_FPGA			BIT(7)
 
-/* Global Frame Length Adjustment Register */
-#define GFLADJ_30MHZ_REG_SEL           (1 << 7)
-#define GFLADJ_30MHZ(n)                        ((n) & 0x3f)
-#define GFLADJ_30MHZ_DEFAULT           0x20
-
 /* Global HWPARAMS7 Register */
 #define DWC3_GHWPARAMS7_RAM1_DEPTH(n)	((n) & 0xffff)
 #define DWC3_GHWPARAMS7_RAM2_DEPTH(n)	(((n) >> 16) & 0xffff)
@@ -1171,8 +1166,6 @@ struct dwc3 {
 	unsigned		has_lpm_erratum:1;
 	unsigned		is_utmi_l1_suspend:1;
 	unsigned		is_fpga:1;
-	unsigned                needs_fifo_resize:1;
-	unsigned                configure_gfladj:1;
 	unsigned		pending_events:1;
 	unsigned		pullups_connected:1;
 	unsigned		setup_packet_pending:1;
-- 
2.17.0

